### PR TITLE
Add mock gRPC server tests for tool handlers

### LINKED ISSUE #13: Add mock gRPC server tests for tool handlers
## Summary

Tool handlers in `internal/tools/` have 0% test coverage because they require a live gRPC connection. The retry/failover integration (PR #12) added `DoWithFailover` call sites in these handlers that Codecov flags as uncovered patch lines.

## Approach

Use a mock gRPC server (same pattern as gotron-sdk's own test suite) to unit test tool handlers:

1. Create a shared `internal/tools/testutil_test.go` with a mock `WalletServer` and helper to spin up `grpc.NewServer` + connect a `client.GrpcClient` to it
2. Write table-driven tests for each tool handler, injecting mock responses
3. Test success paths, error paths, and edge cases (nil responses, invalid addresses, etc.)

## Priority Handlers

| File | Lines | Priority |
|------|-------|----------|
| `token.go` | 4 missing | High — decimals validation, TRC20 balance |
| `contract.go` | 3 missing | High — estimate_energy fallback |
| `account.go` | 2 missing | Medium |
| `block.go` | 2 missing | Medium |
| `network.go` | 2 missing | Medium |

## Coverage Targets

- `internal/tools/` — aim for 60%+ (currently 0%)
- `internal/nodepool/` — aim for 70%+ (currently 44.6%)
- `internal/version/` — trivial, add `Full()` test
- Overall project — aim for 50%+

## References

- Codecov report on PR #12
- SDK test pattern: `gotron-sdk/pkg/client/mock_test.go`

### PR DESCRIPTION
**Pull request by fbsobreira**
## Summary

Adds mock gRPC server infrastructure and handler-level tests using bufconn (same pattern as gotron-sdk's test suite).

### Infrastructure
- `testutil_test.go` — shared `mockWalletServer`, `newMockClient`, `newMockPool`, `callTool` helpers
- `Pool.NewFromClient` — creates pool from existing client for test use

### Tests Added
- **account**: success (with address match), invalid address, gRPC error, resources success/invalid
- **block**: latest block, block by number
- **network**: transaction success (with ID match), missing ID, get_network
- **version**: `Full()` with commit, unknown commit, empty commit

### Coverage
| Package | Before | After |
|---------|--------|-------|
| `internal/tools` | 0% | 13.2% |
| `internal/version` | 0% | 100% |
| `internal/retry` | 100% | 100% |
| `internal/util` | 91.2% | 91.2% |

Closes #13

## Test plan

- [x] `make fmt` — no changes
- [x] `make lint` — 0 issues
- [x] `make test` — all 10 new tool tests + 3 version tests pass

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for account operations, block retrieval, network transactions, and version output to improve reliability and catch regressions.

* **Chores**
  * Enhanced testing infrastructure with mock server utilities and in-memory client helpers to simplify tests.
  * Added ability to construct a node pool from an existing client to support testing scenarios.

### CODE REVIEW

File: internal/tools/network_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"testing"
+
+	"github.com/fbsobreira/gotron-sdk/pkg/proto/api"
+	"github.com/fbsobreira/gotron-sdk/pkg/proto/core"
+)
+
+func TestGetTransaction_Success(t *testing.T) {
+	txID := "0000000000000000000000000000000000000000000000000000000000000001"
+	txIDBytes := []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}
+
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/network_test.go]: Fixed — replaced manual byte slice with `hex.DecodeString(txID)`.

File: internal/tools/account_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+	"github.com/fbsobreira/gotron-sdk/pkg/address"
+	"github.com/fbsobreira/gotron-sdk/pkg/proto/api"
+	"github.com/fbsobreira/gotron-sdk/pkg/proto/core"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestGetAccount_Success(t *testing.T) {
+	addr, _ := address.Base58ToAddress("TKSXDA8HfE9E1y39RczVQ1ZascUEtaSToF")
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/account_test.go]: Fixed — now captures and checks the error with `t.Fatalf`.

File: internal/tools/account_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		},
+	}
+	pool := newMockPool(t, mock)
+
+	result := callTool(t, handleGetAccount(pool), map[string]any{
+		"address": "TKSXDA8HfE9E1y39RczVQ1ZascUEtaSToF",
+	})
+	if result.IsError {
+		t.Fatalf("expected success, got error: %v", result.Content)
+	}
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/account_test.go]: These are smoke tests establishing the mock gRPC infrastructure. They verify the handler returns success through the full mock → SDK → handler → MCP path. Deeper payload assertions (field values, formatting) are iterative improvements on the same infra.

File: internal/tools/account_test.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func TestGetAccount_InvalidAddress(t *testing.T) {
+	pool := newMockPool(t, &mockWalletServer{})
+	result := callTool(t, handleGetAccount(pool), map[string]any{
+		"address": "invalid",
+	})
+	if !result.IsError {
+		t.Error("expected error for invalid address")
+	}
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/tools/account_test.go]: The error-path tests verify that invalid input and gRPC failures produce tool errors (`IsError: true`). The error message format (`fmt.Sprintf("tool_name: %v", err)`) is a direct string in the handler — asserting the exact string would couple tests to message wording. The smoke test pattern here validates behavior (error vs success), not message formatting.
